packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -97,9 +107,13 @@ export const useSliderState = (state: SliderState) => {
 
   const onPointerUp = React.useCallback(
     (ev: React.PointerEvent<HTMLDivElement>): void => {
-      disposables.current.forEach(dispose => dispose());
-      disposables.current = [];
       showStepAnimation();
+      setEventData(prevState => ({
+        ...prevState,
```

Comment:
I don't think you need the function or the spread here since you're overriding all the properties (if you even keep this as the API, see below)